apps/core/middleware.py

```diff
@@ -0,0 +1,17 @@
+from django.shortcuts import get_object_or_404
+from django.utils.translation import get_language_from_path
+
+from apps.core.models import HomePage
+
+
+class ValidateLocaleMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        if lang_code := get_language_from_path(request.path_info):
```

Comment:
Since we are after `LocalMiddleware, we can use `django.utils.translation.get_language()` to get the current language.